src/community/schemaless-features/schemaless-mongo/src/test/java/org/geoserver/schemalessfeatures/mongodb/WMSSchemalessMongoTest.java

```diff
@@ -260,4 +264,35 @@ private void checkStationFeature(JSONObject station) {
         assertNotNull(measurements);
         assertTrue(measurements.size() > 0);
     }
+
+    @Test
+    public void testSchemalessLayerInCapabilities() throws Exception {
+        LayerInfo layerInfo =
+                getCatalog().getLayerByName(new NameImpl("gs", StationsTestSetup.COLLECTION_NAME));
+        // execute the WMS GetMap request
+        Document doc = getAsDOM("wms?request=GetCapabilities&service=WMS&version=1.1.1");
+        XpathEngine xpath = XMLUnit.newXpathEngine();
+        assertTrue(
+                xpath.getMatchingNodes("//Layer/Name[contains(.,geoJSONStations)]", doc).getLength()
+                        > 0);
+    }
+
+    @Test
+    public void testLayerWithNameDifferentFromMongoCollection() throws Exception {
+        String layerRename = "StationsRenamed";
+        Catalog cat = getCatalog();
+        FeatureTypeInfo fti = cat.getFeatureTypeByName("gs", StationsTestSetup.COLLECTION_NAME);
+        fti.setName(layerRename);
+        cat.save(fti);
+        // execute the WMS GetMap request
+        MockHttpServletResponse result =
+                getAsServletResponse(
+                        "wms?SERVICE=WMS&VERSION=1.1.1"
```

Comment:
Please add a WMS 1.3.0 test (Optional, if you see it doesn't take too much time :) )

Reply:
added